### ISSUE TITLE
Option to render proper scrollbar

### ISSUE THREAD

**Issue by jez9999** [Microsoft/vscode#15106]
The current VS Code scrollbar is OK, but I would prefer:
- The scrollbar not to disappear when the mouse stops moving
- The scrollbar to be a bit thicker, thus easier to click
- An up and a down arrow at the top and bottom to scroll one line per click

Maybe a setting could be introduced to render this more "traditional" type of scrollbar.

**Comment by astromediaonly** [comment]
Yeah, this is realy needed. There seems to be options to render arrows atleast but they dont seems to work now (arrows are rendered but clicking does nothing). Im using notebook with touchpad for programming half of the time and without arrow buttons its realy hard to use it - not everybody has scroll wheel.

**Comment by wysisoft** [comment]
I would really like scrollbar up and down buttons too

**Comment by usernamehw** [comment]
There is a setting to change editor's scrollbar width:
```json
"editor.scrollbar.verticalScrollbarSize": 30,
```

**Comment by kdawg1406** [comment]
Having a transparent scrollbar is nuts.  Who thinks editor text that is visible in the scrollbar is a good thing?

There is no option that I can find to get the editor to work like, well, ALL other editors...

**Comment by AdmiralAdama99** [comment]
I am also interested in this feature, particularly being able to click above/below the scroll box location and having it do page up/page down instead of jump. It is bugging me enough that I googled how to change it, and found this feature request. +1, fully support. Please add!

**Comment by SoundOfMind** [comment]
Another vote.  Please add!

The only way to navigate through some massively long lists is by click and drag with the mouse.  Many times using arrow keys or PgUp/PgDn doesn't work as expected.

**Comment by my-name-is** [comment]
I want the scrollbar & slider to be permanently visible.  It gives an indication of where in the document the viewport is.  To me that's useful all the time, not just when I move the mouse or scrollwheels.

In Windows 10 there's a setting "Automatically hide scrollbars in Windows", which I have switched off.  I'd like it if more apps respected that setting.  (Of course, the setting says "Windows" with a capital W, so might just mean for Windows 10's own apps, rather than all windows. Still, consistency across all apps would be nice.)

**Comment by starball5** [comment]
Related to the buttons, see also #22636. Also [this comment there](https://github.com/microsoft/vscode/issues/22636#issuecomment-287058168) which mentions some undocumented setting for enabling them. But I'm confused about what's going on with the button support. The code seems to still be there for those settings: https://github.com/microsoft/vscode/blob/release/1.84/src/vs/editor/common/config/editorOptions.ts#L2563 but it doesn't seem to take effect. Related on Stack Overflow: [How can I enable arrow buttons in VS Code's scrollbars?](https://stackoverflow.com/q/45263508/11107541)

**Comment by Techniktypi** [comment]
The arrows would be especially helpful on small displays to scroll in single line steps with the mouse, so please (re-?)add this!

Decades ago this was standard in OS GUIs but went away for stylistic reasons.
I have to admit that I just started missing the buttons using VS code on a relatively small notebook screen, but still...

I even found a post that suggests that this omission produces accessibility issues for people with certain disabilities.

But I understand that the VS code is not too happy about fixing issues in their software that actually have their root cause in questionable decisions of certain OS suppliers.

**Comment by artem-kamyshev** [comment]
Upvote. Scrolling in VS Code without arrow buttons is not a pretty experience if you don't have a mouse wheel.

**Comment by hem-rajthakur** [comment]
Up and down arrow buttons are really needed in VS Code Scrollbars. Please add them by default.

**Comment by hem-rajthakur** [comment]
I tried from the links given above by @starball5 and this worked for me in latest VS Code in Windows OS.
In settings I have clicked on "Edit in settings.json" link and it opened the following file
C:\Users\{Your User Name}\AppData\Roaming\Code\User\settings.json

 and I have added the mentioned lines in this file:

`"editor.scrollbar": {
            "verticalHasArrows": true,
            "horizontalHasArrows": true,
            "vertical": "visible"
        }`

"vertical": "visible" is optional in it.

Thanks

**Comment by starball5** [comment]
@hem-rajthakur interesting... doesn't work for me on Ubuntu.

**Comment by 00wasaninsidejob** [comment]
@Techniktypi :
> But I understand that the VS code is not too happy about fixing issues in their software that actually have their root cause in questionable decisions of certain OS suppliers.

Is that how things are done these days, "not my fault, not fixing it."?

Or is that attitude borne of the "we're not modeling their bugs in our code" idea.

**Comment by Techniktypi** [comment]
Hello Dave,I‘m thinking of the latter situation.RegardsTechniktypiAm 01.07.25 um 03:55 schrieb Dave
                
            
            
                
                    Von: "Dave" ***@***.***>Datum: 1. Juli 2025An: "microsoft/vscode" ***@***.***>Cc: "Mention" ***@***.***>,"Techniktypi" ***@***.***>Betreff: Re: [microsoft/vscode] Option to render proper scrollbar (#15106)
                
                
00wasaninsidejob left a comment (microsoft/vscode#15106)

@Techniktypi :

But I understand that the VS code is not too happy about fixing issues in their software that actually have their root cause in questionable decisions of certain OS suppliers.

Is that how things are done these days, "not my fault, not fixing it."?
Or is that attitude borne of the "we're not modeling their bugs in our code" idea.
